controllers/users/users.js

```diff
@@ -9,16 +9,30 @@ const nodemailer = require("nodemailer");
 exports.activateUser = async (req, res) => {
   const { activationKey } = req.params;
   const user = await User.findOne({ activationKey });
-  if(user){
-    if(user.emailVerified) {
+  if (user) {
+    if (user.emailVerified) {
       return res.status(400).json({ msg: messages.KEY_HAS_BEEN_ACTIVATED });
     }
     user.emailVerified = true;
     await user.save();
     return res.status(200).json({ msg: messages.USER_ACTIVATED });
   }
-  return res.status(400).json({ msg: messages.ACTIVATION_KEY_IS_INCORRECT }); 
-}
+  return res.status(400).json({ msg: messages.ACTIVATION_KEY_IS_INCORRECT });
+};
+
+exports.getUser = async (req, res) => {
+  try {
+    const user = await User.findById(req.user.id)
+      .select("-_id")
```

Comment:
I don't think so, that excluding fields by select is a good idea. What if we will have thousand of fields?

Maybe this way will be better
`const user = await User.findOne({ _id: req.user.id }, { firstName: 1, lastName: 1 })'

Reply:
@mras2303 
Yes, but probably we'll want to exclude only unwanted (generated by mongoDB for example: _id/__v) or hight risk data in returning user object and this will be small number of fields. So better exclude by 'select' couple of fields than add thousand of fields which we wanted with your suggestion. At this point we getting 6 fields and excluding 5 so even for now it's fine I guess?